### ISSUE TITLE
Implicit Flow not working as it should?

### ISSUE THREAD

**Issue by WilderPereira** [spring-projects/spring-security-oauth#1249]
I was trying to use Amazon's Cognito as a authentication provider using the Implicit Flow, but I've faced some problems with Spring.

I implemented the Authorization Code using Spring with almost the same identity provider configurations and it works perfectly.

The first problem I've faced when implementing the Implicit Flow is that I get a `NullPointer Exception` when I don't set a AccessTokenUri in my ImplicitResourceDetails for the RestTemplate. The exact same problem was stated in [this issue](https://github.com/spring-projects/spring-security-oauth/issues/412), and @dsyer [commented](https://github.com/spring-projects/spring-security-oauth/issues/412#issuecomment-75523032) that 

> You have to set the "accessTokenUri" not the "userAuthorizationUri" for the implicit resource type (makes sense since that's where you get the token from I guess). That will fix the NPE.

but OAuth2 Implicit Flow doesn't have an access token uri, right?
As I couldn't find another solution for this, I followed his suggestions anyway.

After setting the value of my `accessToken` to what should be the `authorizationUri` I got another error:  
`org.springframework.web.client.HttpClientErrorException: 405 Method Not Allowed`. 

That happens because the [retrieveToken](https://github.com/spring-projects/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/client/token/OAuth2AccessTokenSupport.java#L117) method from `OAuth2AccessTokenSupport` class calls a method that always return HTTP.POST as the request method. This is called from [ImplicitAccessTokenProvider.obtainAccessToken](https://github.com/spring-projects/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/client/token/grant/implicit/ImplicitAccessTokenProvider.java#L55).

This error makes sense since the requests for the Authorization Endpoint should only accept HTTP GET as stated in the [Section 3.1 of RFC 9749](https://tools.ietf.org/html/rfc6749#section-3.1).

After this error I've overridden the [ImplicitAccessTokenProvider](https://github.com/spring-projects/spring-security-oauth/blob/master/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/client/token/grant/implicit/ImplicitAccessTokenProvider.java) class to do a GET instead of POST, but it does't redirect the user to the Identity Provider's login page as it does when using the Authorization Flow.

I'm quite convinced that this is not the best approach for achieving what I want, since I had do many workarounds and still couldn't do it.

**TLDR;**

1. Is there a way of retrieving the token from the Authorization Uri using the Implicit Flow with Spring?

2. Why is the Access Token Uri required when using the Implicit Flow? And why does it do a POST request even when using the ImplicitResourceDetails?

After spending a couple of hours I've realized that this could be an old problem.
http://forum.spring.io/forum/spring-projects/security/oauth/120611-trouble-configuring-implicit-flow
https://github.com/spring-projects/spring-security-oauth/issues/412#issuecomment-75523032

**Comment by nagano** [comment]
Any news on this issue?
